app/robots.ts

```diff
@@ -0,0 +1,16 @@
+import type { MetadataRoute } from 'next';
+import { BASE_PATH, BASE_URL } from '@/next.constants.mjs';
+
+// This is the combination of the Application Base URL and Base PATH
+const baseUrlAndPath = `${BASE_URL}${BASE_PATH}`;
+
+const robots = (): MetadataRoute.Robots => ({
+  rules: {
+    userAgent: '*',
+    disallow: ['/dist/', '/docs/'],
+    allow: ['/dist/latest/', '/dist/latest/docs/api/', '/api/'],
+  },
+  sitemap: `${baseUrlAndPath}/sitemap.xml`,
+});
```

Comment:
```suggestion
sitemap: `${baseUrlAndPath}/sitemap.xml`,
host: 'https://nodejs.org'
});
```
We should add host ??

Reply:
I don't think so; it's optional and not mandatory by the robots.txt spec

Reply:
actually on we have robot entry. So I'm ok to remove it but keep it's cost nothing and it's can useful.
REF: https://nodejs.org/robots.txt

Reply:
We do have, but it has no value. It is there because the `next-sitemap` script required it.

I don't see any benefit in having it. It has no value for us, and the search engines already know the host. No website that I know uses that field.

Reply:
gosh the docs are pretty sparse around sitemap... next links to wikipedia... which links to a yahoo article from 2009